### PR TITLE
[Feature] Support MultiDataPreprocessor

### PR DESCRIPTION
**Pull request by Czm369**
image

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet/models/data_preprocessors/data_preprocessor.py [hunk 1]
```diff
@@ -333,3 +334,89 @@ def forward(
                         value=self.seg_pad_value)
 
         return batch_inputs, batch_data_samples
+
+
+@MODELS.register_module()
+class MultiDataPreprocessor(nn.Module):
```
Comment (hhaAndroid) [inline mmdet/models/data_preprocessors/data_preprocessor.py:340]: MultiDataPreprocessor -> MultiBranchDataPreprocessor

File: mmdet/models/data_preprocessors/data_preprocessor.py [hunk 2]
```diff
@@ -333,3 +334,89 @@ def forward(
...
+            training (bool): Whether to enable training time augmentation.
+
+        Returns:
+            Tuple[Dict[torch.Tensor], Dict[Optional[list]]]: Each tuple of
+            zip(dict, dict) is the data in the same format as the model input.
+        """
+
+        if training is False:
+            return self.data_preprocessor(data, training)
+        multi_data = {}
```
Comment (hhaAndroid) [inline mmdet/models/data_preprocessors/data_preprocessor.py:371]: multi_data -> multi_branch_data

File: mmdet/models/data_preprocessors/data_preprocessor.py [hunk 3]
```diff
@@ -333,3 +334,89 @@ def forward(
...
+
+        Args:
+            device (int or torch.device, optional): The desired device of the
+                parameters and buffers in this module.
+
+        Returns:
+            nn.Module: The model itself.
+        """
+
+        self.data_preprocessor._device = torch.device(device)
```
Comment (hhaAndroid) [inline mmdet/models/data_preprocessors/data_preprocessor.py:400]: self.data_preprocessor.to()

File: mmdet/models/data_preprocessors/data_preprocessor.py [hunk 4]
```diff
@@ -333,3 +334,89 @@ def forward(
...
+        return super().to(device)
+
+    def cuda(self, *args, **kwargs) -> nn.Module:
+        """Overrides this method to set the :attr:`device`
+
+        Returns:
+            nn.Module: The model itself.
+        """
+
+        self.data_preprocessor._device = torch.device(
```
Comment (hhaAndroid) [inline mmdet/models/data_preprocessors/data_preprocessor.py:410]: self.data_preprocessor.cuda()

File: mmdet/models/data_preprocessors/data_preprocessor.py [hunk 5]
```diff
@@ -333,3 +334,89 @@ def forward(
...
+        return super().cuda()
+
+    def cpu(self, *args, **kwargs) -> nn.Module:
+        """Overrides this method to set the :attr:`device`
+
+        Returns:
+            nn.Module: The model itself.
+        """
+
+        self.data_preprocessor._device = torch.device('cpu')
```
Comment (hhaAndroid) [inline mmdet/models/data_preprocessors/data_preprocessor.py:421]: self.data_preprocessor.cpu()

**Review by hhaAndroid — DISMISSED** [review]

File: mmdet/models/data_preprocessors/data_preprocessor.py [hunk 6]
```diff
@@ -333,3 +334,85 @@ def forward(
                         value=self.seg_pad_value)
 
         return batch_inputs, batch_data_samples
+
+
+@MODELS.register_module()
+class MultiBranchDataPreprocessor(nn.Module):
```
Comment (ZwwWayne) [inline mmdet/models/data_preprocessors/data_preprocessor.py:340]: This module should inherits from the BaseDataPreprocessor to have compatible functionalities.

**Pull request merged**